### PR TITLE
[WIP] remote http: add exception for no_traverse

### LINKED ISSUE #2380: Fix `cache_exists` for RemoteHTTP
As noted by @Suor  in https://github.com/iterative/dvc/pull/2375
Http remote will not be working properly in case of `no_traverse=False`, due to fact that `list_cache_paths` is not implemented for `RemoteHTTP`.

### PR DESCRIPTION
**Pull request by mroutis**
Fix #2380

* [ ] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [ ] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

### CODE REVIEW

File: tests/func/remote/test_http.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from dvc.main import main
+
+
+def test_incompatibility_with_no_traverse(dvc_repo, caplog):
```
Comment (efiop) [inline tests/func/remote/test_http.py]: Just create a unit test with initializing RemoteHTTP in tests/unit/remote/test_http.py

File: tests/func/test_repro.py [hunk 2]
```diff
@@ -1182,6 +1182,8 @@ def test(self):
             self.assertEqual(ret1, 0)
             self.assertEqual(ret2, 0)
 
+            main(["remote", "modify", "mycache", "no_traverse", "false"])
```
Comment (efiop) [inline tests/func/test_repro.py]: This is not needed.

File: tests/func/remote/test_http.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    main(["remote", "add", "http", "https://example.com"])
+
+    assert 0 != main(["status", "--remote", "http"])
+
+    assert (
```
Comment (efiop) [inline tests/func/remote/test_http.py]: This test is incorrect, no_traverse is True by default, which is supported by HTTP.

File: dvc/remote/http.py [hunk 4]
```diff
@@ -42,6 +42,12 @@ def __init__(self, repo, config):
         url = config.get(Config.SECTION_REMOTE_URL)
         self.path_info = self.path_cls(url) if url else None
 
+        if not self.no_traverse:
+            raise DvcException(
+                "RemoteHTTP does not support 'no_traverse' option. "
+                "Disable it with `dvc remote modify <name> no_traverse false`"
```
Comment (efiop) [inline dvc/remote/http.py]: Also please don't forget to fix the error message.

**Comment by mroutis** [comment]
Closing in favor of another PR

**Pull request closed without merging**